CONTRIBUTING.md

```diff
@@ -1,32 +1,101 @@
+####INDEX
+* [Pull Requests](#pull-requests)
+* [Setting Up Your Environment](#setting-up-your-environment)
+  * [Assumptions](#assumptions)
+  * [Your Repository](#setting-up-your-repository)
+  * [Solution Dependencies](#setting-up-the-solution-dependencies)
+
 Pull Requests
 =============
 
-* All PRs must be related to an open issue
-* If a PR changes the scripting API, it must include all related changes to the documentation. 
+* Please reference [Style Guide](STYLEGUIDE.md)
+* All PRs must be related to an open issue.  If there isn't already an issue
+  that applies to your PR, please submit a new issue with the PR.
+* If a PR changes the scripting API, it must include all related changes to the documentation.
```

Comment:
Maybe append the sentence: "The documentation is edited by changing or adding to the *.rst files under /doc/source."